### PR TITLE
Bound RewardPointsin pallet-staking

### LINKED ISSUE #12039: Bound `RewardPoints`in pallet-staking
... and add a log every time a validators is rewarded. 

This is a very low hanging fruit in the direction of bounding the staking pallet i.e. https://github.com/paritytech/substrate/issues/8629

### PR DESCRIPTION
**Pull request by Doordashcon**
previously #12039

### CODE REVIEW

File: bin/node/runtime/src/lib.rs [hunk 1]
```diff
@@ -569,6 +569,7 @@ impl pallet_staking::Config for Runtime {
 	type OnStakerSlash = NominationPools;
 	type WeightInfo = pallet_staking::weights::SubstrateWeight<Runtime>;
 	type BenchmarkingConfig = StakingBenchmarkingConfig;
+	type MaxRewardPoints = ConstU32<300>;
```
Comment (kianenigma) [inline bin/node/runtime/src/lib.rs:572]: The number of keys in this map is the number of validators -- staking should have a single `MaxValidators`, and use that as the bound for this type as well.

Comment (Doordashcon) [reply bin/node/runtime/src/lib.rs:572]: was wondering about that, thanks!

**Review by kianenigma — COMMENTED** [review]
Looks mostly good, I will review closer later.

**Comment by kianenigma** [comment]
overlaps with https://github.com/paritytech/substrate/pull/11967

File: frame/staking/src/lib.rs [hunk 2]
```diff
@@ -368,17 +372,47 @@ pub struct ActiveEraInfo {
...
 	fn default() -> Self {
-		EraRewardPoints { total: Default::default(), individual: BTreeMap::new() }
+		EraRewardPoints { total: Default::default(), individual: BoundedBTreeMap::new() }
+	}	
+}
+
+impl<T: Config> EraRewardPoints<T> {
+	fn add_points(&mut self, validator: T::AccountId, reward_points: u32) -> Result<(), Error<T>> {
+		let mut points = self.individual.get(&validator).map(|x| *x).unwrap_or_default();
+		points += reward_points;
```
Comment (kianenigma) [inline frame/staking/src/lib.rs:395]:
```suggestion
		points = points.saturating_add(reward_points);
```

**Comment by Doordashcon** [comment]
@kianenigma there is still panicking code in the runtime, I am working on it

**Comment by kianenigma** [comment]
@Doordashcon see https://github.com/paritytech/substrate/issues/9724#issuecomment-1236756780.

**Pull request closed without merging**